### PR TITLE
Add a cookie rotator to migrate to SHA256

### PR DESCRIPTION
**Pull request by segiddins**
Following https://guides.rubyonrails.org/v7.0/upgrading_ruby_on_rails.html#key-generator-digest-class-changing-to-use-sha256

### CODE REVIEW

File: config/initializers/cookie_rotator.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Rails.application.config.after_initialize do
+  Rails.application.config.action_dispatch.cookies_rotations.tap do |cookies|
+    salt = Rails.application.config.action_dispatch.authenticated_encrypted_cookie_salt
+    secret_key_base = Rails.application.secret_key_base
+
+    key_generator = ActiveSupport::KeyGenerator.new(
+      secret_key_base, iterations: 1000, hash_digest_class: OpenSSL::Digest::SHA1
```
Comment (jchestershopify) [inline config/initializers/cookie_rotator.rb:7]: Why is this SHA1 but the other setting is SHA256?

Comment (simi) [reply config/initializers/cookie_rotator.rb:7]: If I understand it well, this is to keep it compatible. On this line it is setting up back the old way (SHA1, https://github.com/rails/rails/blob/1fd0bc1094144b268a3b931a92eb6aeb32fe5c7b/guides/source/configuring.md#configactive_supportkey_generator_hash_digest_class) and later `key_generator_hash_digest_class` is configured to use the new way (SHA256).

That way old cookies are "rotated" when reaches the app without invalidation. They are just migrated from SHA1 to SHA256. We can remove this rotator after some time.

**Comment by simi** [comment]
Migration seems ok (and could be tested on staging before deploy). But this worries me a little.

> [2.11 Digest class for ActiveSupport::Digest changing to SHA256](https://guides.rubyonrails.org/v7.0/upgrading_ruby_on_rails.html#digest-class-for-activesupport-digest-changing-to-sha256)
The default digest class for ActiveSupport::Digest is changing from SHA1 to SHA256. This has consequences for things like Etags that will change and cache keys as well. Changing these keys can have impact on cache hit rates, so be careful and watch out for this when upgrading to the new hash.

Any idea `Etags` are used for Fastly caching and is cache hit ratio going to be affected? There were already some troubles with this (resulting into high DB load) recently. :thinking:

**Comment by simi** [comment]
@segiddins any idea about the raised concerns? Do you think we can check on staging somehow?

**Comment by segiddins** [comment]
I haven't gotten back around to this. I'm fairly certain I can debug locally

**Comment by simi** [comment]
I did some testing and it seems safe.

**Pull request merged**